cocos2d/core/labelttf/CCLabelTTFCanvasRenderCmd.js

```diff
@@ -92,43 +74,23 @@ cc.LabelTTF._firsrEnglish = /^[a-zA-Z0-9ÄÖÜäöüßéèçàùêâîôû]/;
         return this._fontClientHeight;
     };
 
-    proto._updateTexture = function () {
-        this._dirtyFlag = this._dirtyFlag & cc.Node._dirtyFlags.textDirty ^ this._dirtyFlag;
-        var node = this._node;
-        var locContext = this._getLabelContext(), locLabelCanvas = this._labelCanvas;
-        var locContentSize = node._contentSize;
-
-        if (node._string.length === 0) {
-            locLabelCanvas.width = 1;
-            locLabelCanvas.height = locContentSize.height || 1;
-            node._texture && node._texture.handleLoadedTexture();
-            node.setTextureRect(cc.rect(0, 0, 1, locContentSize.height));
-            return true;
-        }
-
-        //set size for labelCanvas
-        locContext.font = this._fontStyleStr;
-        this._updateTTF();
-        var width = locContentSize.width, height = locContentSize.height;
-        var flag = locLabelCanvas.width === width && locLabelCanvas.height === height;
-        locLabelCanvas.width = width;
-        locLabelCanvas.height = height;
-        if (flag) locContext.clearRect(0, 0, width, height);
-
-        //draw text to labelCanvas
-        this._drawTTFInCanvas(locContext);
-        node._texture && node._texture.handleLoadedTexture();
-
-        node.setTextureRect(cc.rect(0, 0, width, height));
-        return true;
+    proto._updateColor = function(){
+        this._setColorsString();
+        this._updateTexture();
     };
 
-    proto._measureConfig = function () {
-        this._getLabelContext().font = this._fontStyleStr;
-    };
+    proto._setColorsString = function () {
```

Comment:
Since the function mostly used in init or setX function ,I don't think it should be changed